coreblocks/peripherals/axi_lite.py

```diff
@@ -0,0 +1,283 @@
+from transactron import Method, def_method, TModule
+from transactron.core import Transaction
+from transactron.lib.connectors import Forwarder
+from enum import Enum, unique
+
+
+class AXILiteParameters:
+    """Parameters of the AXI-Lite bus.
+
+    Parameters
+    ----------
+    data_width: int
+        Width of "data" signals for "write data" and "read data" channels. Must be either 32 or 64 bits. Defaults to 64
+    addr_width: int
+        Width of "addr" signals for "write address" and "read address" channels. Defaults to 64 bits.
+    """
+
+    def __init__(self, *, data_width=64, addr_width=64):
+        self.data_width = data_width
+        self.addr_width = addr_width
+
+
+class AXILiteLayout:
+    """AXI-Lite bus layout generator
+
+    Parameters
+    ----------
+    axil_params: AXILiteParameters
+        Patameters used to generate AXI-Lite layout
+    master: Boolean
+        Whether the layout should be generated for master side
+        (if false it's generatd for the slave side)
+
+    Attributes
+    ----------
+    axil_layout: Record
+        Record of a AXI-Lite bus.
+    """
+
+    def __init__(self, axil_params, *, master=True):
+        write_address = [
+            ("val", 1, DIR_FANOUT if master else DIR_FANIN),
+            ("rdy", 1, DIR_FANIN if master else DIR_FANOUT),
+            ("addr", axil_params.addr_width, DIR_FANOUT if master else DIR_FANIN),
+            ("prot", 3, DIR_FANOUT if master else DIR_FANIN),
+        ]
+
+        write_data = [
+            ("val", 1, DIR_FANOUT if master else DIR_FANIN),
+            ("rdy", 1, DIR_FANIN if master else DIR_FANOUT),
+            ("data", axil_params.data_width, DIR_FANOUT if master else DIR_FANIN),
+            ("strb", axil_params.data_width // 8, DIR_FANOUT if master else DIR_FANIN),
+        ]
+
+        write_response = [
+            ("val", 1, DIR_FANIN if master else DIR_FANOUT),
+            ("rdy", 1, DIR_FANOUT if master else DIR_FANIN),
+            ("resp", 2, DIR_FANIN if master else DIR_FANOUT),
+        ]
+
+        read_address = [
+            ("val", 1, DIR_FANOUT if master else DIR_FANIN),
+            ("rdy", 1, DIR_FANIN if master else DIR_FANOUT),
+            ("addr", axil_params.addr_width, DIR_FANOUT if master else DIR_FANIN),
+            ("prot", 3, DIR_FANOUT if master else DIR_FANIN),
+        ]
+
+        read_data = [
+            ("val", 1, DIR_FANIN if master else DIR_FANOUT),
+            ("rdy", 1, DIR_FANOUT if master else DIR_FANIN),
+            ("data", axil_params.data_width, DIR_FANIN if master else DIR_FANOUT),
+            ("resp", 2, DIR_FANIN if master else DIR_FANOUT),
+        ]
+
+        self.axil_layout = [
+            ("write_address", write_address),
+            ("write_data", write_data),
+            ("write_response", write_response),
+            ("read_address", read_address),
+            ("read_data", read_data),
+        ]
+
+
+@unique
+class States(Enum):
```

Comment:
This is a single state, not a set of states.
```suggestion
class State(Enum):
```

Reply:
After changing `state_machine_request` to use Amaranth's FSM this class is obsolete and was removed